google-cloud-examples/src/test/java/com/google/cloud/examples/storage/snippets/ITBlobSnippets.java

```diff
@@ -106,74 +120,232 @@ public void after() {
+    System.setOut(systemOut);
+    assertTrue(snippetOutput.contains("Bucket: " + remoteBlob.getBucket()));
+    assertTrue(snippetOutput.contains("Bucket: " + remoteBlob.getBucket()));
+    assertTrue(snippetOutput.contains("CacheControl: " + remoteBlob.getCacheControl()));
+    assertTrue(snippetOutput.contains("ComponentCount: " + remoteBlob.getComponentCount()));
+    assertTrue(snippetOutput.contains("ContentDisposition: " + remoteBlob.getContentDisposition()));
+    assertTrue(snippetOutput.contains("ContentEncoding: " + remoteBlob.getContentEncoding()));
+    assertTrue(snippetOutput.contains("ContentLanguage: " + remoteBlob.getContentLanguage()));
+    assertTrue(snippetOutput.contains("ContentType: " + remoteBlob.getContentType()));
+    assertTrue(snippetOutput.contains("Crc32c: " + remoteBlob.getCrc32c()));
+    assertTrue(snippetOutput.contains("Crc32cHexString: " + remoteBlob.getCrc32cToHexString()));
+    assertTrue(snippetOutput.contains("ETag: " + remoteBlob.getEtag()));
+    assertTrue(snippetOutput.contains("Generation: " + remoteBlob.getGeneration()));
+    assertTrue(snippetOutput.contains("Id: " + remoteBlob.getBlobId()));
+    assertTrue(snippetOutput.contains("KmsKeyName: " + remoteBlob.getKmsKeyName()));
+    assertTrue(snippetOutput.contains("Md5Hash: " + remoteBlob.getMd5()));
+    assertTrue(snippetOutput.contains("Md5HexString: " + remoteBlob.getMd5ToHexString()));
+    assertTrue(snippetOutput.contains("MediaLink: " + remoteBlob.getMediaLink()));
+    assertTrue(snippetOutput.contains("Metageneration: " + remoteBlob.getMetageneration()));
+    assertTrue(snippetOutput.contains("Name: " + remoteBlob.getName()));
+    assertTrue(snippetOutput.contains("Size: " + remoteBlob.getSize()));
+    assertTrue(snippetOutput.contains("StorageClass: " + remoteBlob.getStorageClass()));
+    assertTrue(snippetOutput.contains("TimeCreated: " + new Date(remoteBlob.getCreateTime())));
+    assertTrue(
+        snippetOutput.contains("Last Metadata Update: " + new Date(remoteBlob.getUpdateTime())));
+    assertTrue(snippetOutput.contains("temporaryHold: disabled"));
+    assertTrue(snippetOutput.contains("eventBasedHold: disabled"));
+    assertTrue(snippetOutput.contains("User metadata:"));
+    assertTrue(snippetOutput.contains("k=v"));
+  }
+
+  @Test
+  public void testListObjects() {
+    final ByteArrayOutputStream snippetOutputCapture = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(snippetOutputCapture));
+    ListObjects.listObjects(PROJECT_ID, BUCKET);
+    String snippetOutput = snippetOutputCapture.toString();
+    assertTrue(snippetOutput.contains(BLOB));
+    System.setOut(System.out);
+  }
+
+  @Test
+  public void testListObjectsWithPrefix() {
+    final ByteArrayOutputStream snippetOutputCapture = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(snippetOutputCapture));
+    storage.create(BlobInfo.newBuilder(BlobId.of(BUCKET, "/a/1.txt")).build());
```

Comment:
Recommend removing the forward slash for these names.

Reply:
I used them to be consistent with samples in other languages, for example see the Go/Ruby/Python samples here: https://cloud.google.com/storage/docs/listing-objects#storage-list-objects-nodejs

Reply:
I updated Python and working through the others. I recommend staying away from '/a/1.txt' if possible as has caused user confusion.